modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2142,23 +2143,35 @@ def applyier(df, internal_indices, other=[], internal_other_indices=[]):
     # END Map across rows/columns
 
     # __getitem__ methods
+    __getitem_bool = Binary.register(
+        lambda df, r: df[r], join_type="left", labels="drop"
+    )
+
+    def __validate_bool_indexer(self, indexer):
+        if len(indexer) != len(self.index):
+            raise ValueError(
+                f"Item wrong length {len(indexer)} instead of {len(self.index)}."
+            )
+        if isinstance(indexer, pandas.Series) and not indexer.equals(self.index):
+            warnings.warn(
+                "Boolean Series key will be reindexed to match DataFrame index.",
+                PendingDeprecationWarning,
+                stacklevel=4,
+            )
+
     def getitem_array(self, key):
-        # TODO: dont convert to pandas for array indexing
         if isinstance(key, type(self)):
```

Comment:
qq: what does it mean to be checking for whether the key is of type self. Are we checking to see if we're looking up a query_compiler?

Reply:
Yep, we're looking for another pandas query compiler as opposed to, say, a list. It's only when we have another pandas query compiler that we can do the getitem as a binary operation.

Reply:
this should mostly happen when we index with a bool-like Series, e.g. kind of `df[df.col == 5]` lines